ckan/controllers/package.py

```diff
@@ -343,29 +341,21 @@ def read(self, id):
                    'user': c.user, 'for_view': True,
                    'auth_user_obj': c.userobj}
         data_dict = {'id': id, 'include_tracking': True}
-
-        # interpret @<revision_id> or @<date> suffix
-        split = id.split('@')
-        if len(split) == 2:
-            data_dict['id'], revision_ref = split
-            if model.is_id(revision_ref):
-                context['revision_id'] = revision_ref
-            else:
-                try:
-                    date = h.date_str_to_datetime(revision_ref)
-                    context['revision_date'] = date
-                except TypeError, e:
-                    abort(400, _('Invalid revision format: %r') % e.args)
-                except ValueError, e:
-                    abort(400, _('Invalid revision format: %r') % e.args)
-        elif len(split) > 2:
-            abort(400, _('Invalid revision format: %r') %
-                  'Too many "@" symbols')
+        activity_id = request.params.get('activity_id')
 
         # check if package exists
         try:
             c.pkg_dict = get_action('package_show')(context, data_dict)
             c.pkg = context['package']
+
+            if activity_id:
+                c.pkg_dict = context['session'].query(model.Activity).get(
+                    activity_id
+                ).data['package']
+                # Don't crash on old activity records, which do not include
+                # resources or extras.
+                c.pkg_dict.setdefault('resources', [])
+                c.is_activity_archive = True
```

Comment:
We are actively trying to avoiding setting new things into `c` (or Flask's `g`), can you pass this explicitly as an extra var?

Reply:
ckanext-scheming and any site that uses multiple IDatasetForm plugins use this value, but it's not used by the core templates because they only support a single type

Reply:
I've addressed this in #3972 now